Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -335,7 +343,57 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
 
     // MARK: CompleteOnSessionCheckout
 
-    // TODO: Call .completeOnSessionCheckout when self.confirmPaymentSuccessfulProperty.signal is called
+    let completeCheckoutWithExistingCardInput: Signal<GraphAPI.CompleteOnSessionCheckoutInput, Never> = Signal
+      .combineLatest(self.confirmPaymentSuccessfulProperty.signal.skipNil(), checkoutId, selectedExistingCard)
```

Comment:
I think this is going to have the same issue as before. Let's say I first select an existing card, then add a new card. Now both `selectedExistingCard` and `selectedNewCard` will have values set (the existing one is outdated, but it doesn't know that). Once `confirmPaymentSuccessfulProperty` has an updated value, both this and the new card version of it will fire. I'm pretty sure that, in order to prevent this from happening, we need to add a `filter` _after_ the `combineLatest`.

Alternatively, if we want to be able to just use `selectedNewCard`/`selectedExistingCard` we could map the opposite version to `nil` and filter on that (or return early/map to nil in the map/ect). Ie make sure that these signals won't both have valid cards at the same time.

For this specific call, though, I think it'd be easiest to just have one `completeCheckoutWithCardInput` method and switch on `card.isNewPaymentMethod` for the one field that's different.